keps/sig-cloud-provider/2436-controller-manager-leader-migration/README.md

```diff
@@ -349,18 +354,131 @@ feature without providing a configuration, the default configuration will reflec
+
+This feature can be re-enabled without side effects.
+
 ###### Are there any tests for feature enablement/disablement?
 
 Yes. Unit & integration tests include flag/configuration parsing. E2E test will have cases with the feature enabled and
 disabled.
 
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The rollout may fail if the configuration file does not represent correct controller-to-manager.
+This can cause controllers referred in the configuration file to either be unavailable or run in multiple instances.
+
+The rollback may fail if the leader election of the controller manager is not properly configured.
+For example, multiple instances of the same controller manager are running without election, or none of the instances become the leader.
+In these situations, all controllers will be either unavailable or conflict among multiple instances.
+
+###### What specific metrics should inform a rollback?
+
+If neither controller managers show `leader_active` for the main leader lock or the migration lock, Leader Migration may fail to activate and thus needs rollback.
+If any of the controllers indicate they are unavailable through their per-controller metric, Leader Migration may need reconfiguration.
+The metrics of each controller are specific to the implementation of each cloud provider and out of scope of this KEP,
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+The manual testing showed a clean takeover of the migration leader during both upgrade and downgrade process.
+This process will be tested as part of the e2e suite, required by [Graduation Criteria](#graduation-criteria).
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+None.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+N/A. This feature is never used by any user workloads.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details: The cluster administrator SHOULD have access to logs and metrics during cluster upgrade.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A. This feature is short-lived and only active during cluster upgrade.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Metrics
+  - leader_active
+  - other per-controller availability metrics.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+It would help if every controller that the controller manager hosts expose metrics about their availability.
+However, per-controller metrics are out of scope of this KEP.
```

Comment:
A metric for whether a lease(s) is current or expired would improve observability.

Reply:
A expired lease will cause controller managers to `FATAL` instantly. The metric would not be visible because by the time lease transit to "expired" state, the process exit before promethues could scape the new value.

Reply:
Yes & no. The metric for this does not have to come from the Controller-Manager holding the lock. We could enhance the client code to emit this metric for any lease it was attempting to acquire. In that case the both the current holder and the clients attempting to acquire the lock would show it as current and the clients attempting to acquire would not it as expired. If the current is dead and there was no one attempting to acquire, then you would get nothing but that itself is a signal. Worth noting this is asking if there are signals which would help not a promise to implement those signals.

Reply:
Oh I see this is the "good to have" metrics, not metrics we need to implement. I am adding lease-related metrics here. Even though I do not see the `Lease` resource exporting any resource-level metrics yet.

Reply:
Updated.